cpp/perspective/src/cpp/base.cpp

```diff
@@ -280,6 +312,137 @@ filter_op_to_str(t_filter_op op) {
     return "";
 }
 
+t_filter_op
+str_to_filter_op(std::string str) {
+    if (str == "<") {
```

Comment:
don't like switch statements?

Reply:
You can’t switch on a string in C++, so it was this or write another enum that gets mapped to the original enum.

Reply:
you can do some [stuff](http://www.rioki.org/2016/03/31/cpp-switch-string.html), but I think what you've done here is more readable and fine.